sklearn/datasets/_openml.py

```diff
@@ -722,6 +821,21 @@ def fetch_openml(
         Missing values in the 'data' are represented as NaN's. Missing values
         in 'target' are represented as NaN's (numerical target) or None
         (categorical target)
+
+    Notes
+    -----
+    It should be noted that the `"pandas"` and `"liac-arff"` parsers can lead
+    to different data types in the output. The notable differences are the
+    following:
+
+    - The `"liac-arff"` parser always considers nominal features as string.
+      To the contrary, the `"pandas"` parser instead infers the type while
+      reading and numerical categories will be casted into integers or floats
+      whenever possible.
+    - The `"liac-arff"` parser is using floats to encode numerical features
+      tagged as 'REAL' and 'NUMERICAL' in the metadata. To the contrary, the
+      `"pandas"` parser instead infers if these numerical features corresponds
+      to integers.
```

Comment:
```suggestion
    - The `"liac-arff"` parser uses floats to encode numerical features
      tagged as 'REAL' and 'NUMERICAL' in the metadata. The `"pandas"`
      parser instead infers if these numerical features corresponds
      to integers.
```

If an attribute is explicitly marked as READ, we should probably tell pandas to use a float64 dtype explicitly for that column instead, no?

It there an explicit INTEGER tag in the ARFF metadata spec?